airsonic-main/src/main/java/org/airsonic/player/service/MediaScannerService.java

```diff
@@ -53,7 +66,7 @@
 
     private MediaLibraryStatistics statistics;
 
-    private boolean scanning;
+    private volatile boolean scanning;
```

Comment:
set to volatile so all threads can see it from main memory (can also make it atomic, but isn't necessary)